client/src/Component/ViewNotes.js

```diff
@@ -0,0 +1,155 @@
+import React, { Component } from "react";
+import { withRouter } from "react-router-dom";
+import { connect } from "react-redux";
+import { addNote, updateNote, deleteNote } from "../Actions";
+
+import Notes from "./Notes";
+
+class ViewNotes extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: "",
+      content: "",
+      id: -1
+    };
+  }
+
+  componentDidMount() {
+    if (this.props.notes.length > 0) {
+      const defaultNote = this.props.notes[0];
+      this.setState({
+        title: defaultNote.title,
+        content: defaultNote.content,
+        id: defaultNote._id
+      });
+    }
+  }
+
+  addNoteToggle = () => {
+    this.setState({
+      title: "",
+      content: "",
+      id: -1
+    });
+  };
+
+  noteChangeHandler = event => {
+    const { name, value } = event.target;
+    this.setState({
+      [name]: value
+    });
+  };
+
+  updateNote = event => {
+    event.preventDefault();
+    const editNote = {
+      id: this.state.id,
+      title: this.state.title,
+      content: this.state.content
+    };
+    const note = {
+      title: this.state.title,
+      content: this.state.content
+    };
+
+    if (this.state.content.length === 0 || this.state.title.length === 0) {
+      alert(
+        "You need to have a title and Body in your note before you can save"
```

Comment:
Adding validations is often an afterthought. That you've gone to this level of detail is really impressive to me. Well done.